saleor/graphql/product/tests/test_variant.py

```diff
@@ -6083,3 +6083,77 @@ def test_product_variant_deactivate_preorder_as_app(
     )
 
     assert_no_permission(response)
+
+
+VARIANT_CREATE_MUTATION = """
+    mutation variantCreate($productId: ID!, $attrId: ID!) {
+        productVariantCreate (
+            input: {
+                sku: "my-sku",
+                product: $productId,
+                attributes: [{id: $attrId, values:["1"]}]
+            }
+        )
+        {
+            productVariant {
+                id
+            }
+            errors {
+                field,
+                message,
+                code,
+                attributes
+            }
+        }
+    }
+"""
+
+
+def test_variant_create_product_without_variant_attributes(
+    product_with_product_attributes, staff_api_client, permission_manage_products
+):
+    product = product_with_product_attributes
+
+    prod_id = graphene.Node.to_global_id("Product", product.pk)
+    attr_id = graphene.Node.to_global_id(
+        "Attribute", product.product_type.product_attributes.first().pk
+    )
+
+    response = staff_api_client.post_graphql(
+        VARIANT_CREATE_MUTATION,
+        variables={"productId": prod_id, "attrId": attr_id},
+        permissions=[permission_manage_products],
+    )
+    content = get_graphql_content(response)
+
+    errors = content["data"]["productVariantCreate"]["errors"]
+    assert errors
+    assert errors[0]["code"] == ProductErrorCode.ATTRIBUTE_CANNOT_BE_ASSIGNED.name
+    assert len(errors[0]["attributes"]) == 1
+    assert errors[0]["attributes"][0] == attr_id
+
+
+def test_variant_create_product_with_variant_attributes_variant_flag_false(
```

Comment:
This test is unnecessary. IMO we shouldn't be able to create such a product type by API. If it's possible, it's a bug.